src/webgpu/util/texture/texel_data.ts

```diff
@@ -355,24 +404,28 @@ export const kTexelRepresentationInfo: {
         [TexelComponent.A]: { dataType: 'unorm', bitLength: 2 },
       },
       encode: components => {
-        components = applyEach(
-          (n: number) => (floatAsNormalizedInteger(n, 10, false), n),
-          kRGB
-        )(components);
-        components = applyEach((n: number) => (floatAsNormalizedInteger(n, 2, false), n), [
-          TexelComponent.A,
-        ])(components);
-        return components;
+        assert(components.R !== undefined);
+        assert(components.G !== undefined);
+        assert(components.B !== undefined);
+        assert(components.A !== undefined);
+        return {
+          [TexelComponent.R]: floatAsNormalizedInteger(components.R, 10, false),
```

Comment:
This is silly but maybe:
```suggestion
          [TexelComponent.R]: floatAsNormalizedInteger(components.R ?? unreachable(), 10, false),
```

Reply:
Kinda like it. Done.